app/mailers/app_mailer.rb

```diff
@@ -2,21 +2,22 @@ class AppMailer < ActionMailer::Base
 
   default from: 'admin@myflix.com'
 
-  def send_mail_on_register(user)
-    @user = user
-    mail to: user.email, subject: 'You have registered successfully!'
+  def send_mail_on_register(user_id)
+    @user = User.find(user_id)
+    mail to: @user.email, subject: 'You have registered successfully!'
   end
 
-  def send_mail_on_reset_password(user)
-    @user = user
-    @link = reset_password_url(user.reset_password_token, Rails.application.config.action_mailer.default_url_options)
-    mail to: user.email, subject: 'Reset your password now'
+  def send_mail_on_reset_password(user_id)
+    @user = User.find(user_id)
+    @link = reset_password_url(@user.reset_password_token, Rails.application.config.action_mailer.default_url_options)
+    mail to: @user.email, subject: 'Reset your password now'
```

Comment:
I'm not sure whether I really understand what you mean. Do you mean that I should change the default setting to something like
```
default from: Rails.application.secrets.default_mail_from
```
and
drop the line 
```
default from: 'admin@myflix.com'
```
and add ```from:``` key to the ```mail``` call, respectively.
Right?
Do I need a new staging-specific block in secrets.yml?? There is no such item so far. Have a look at secrets.yml.